www/js/services/indicoApiService.js

```diff
@@ -0,0 +1,181 @@
+
+      authBrowser.addEventListener('loadstart', e => {
+        if (e.url.indexOf(redirectUri) == 0) {
+          authBrowser.close();
+          // clear session cookie from browser so it's not sent on further fetches
+          window.cookies.clear();
+          resolve(e.url);
+        }
+      });
+
+      authBrowser.addEventListener('loaderror', err => {
+        authBrowser.close();
+        reject(err);
+      });
+    });
+
+  const authImplicitFlow = server => {
+    return new Promise(async (resolve, reject) => {
+      const authUrl = buildUrl(server.authUrl, {
+        client_id: server.clientId,
+        response_type: 'token',
+        redirect_uri: server.callbackUrl,
+        scope: server.scope,
+      });
+
+      try {
+        const authBrowserResult = await browserSignOn(authUrl.href, server.callbackUrl);
+        const {access_token: accessToken} = getSearchParams(authBrowserResult, '#');
+
+        resolve({...server, token: accessToken});
+      } catch (e) {
+        reject(e);
+      }
+    });
+  };
+
+  const authPkceFlow = server => {
+    return new Promise(async (resolve, reject) => {
+      const state = generateRandomString(128);
+      const codeVerifier = generateRandomString(128);
+      const codeChallenge = await generateCodeChallenge(codeVerifier);
+      const authUrl = buildUrl(server.authUrl, {
+        response_type: 'code',
+        client_id: server.clientId,
+        redirect_uri: server.callbackUrl,
+        state,
+        scope: server.scope,
+        code_challenge: codeChallenge,
+        code_challenge_method: 'S256',
+      });
+
+      try {
+        const authBrowserResult = await browserSignOn(authUrl.href, server.callbackUrl);
+        const {code, state: responseState, error} = getSearchParams(authBrowserResult);
+
+        if (error || !code || state !== responseState) reject();
+
+        const tokenResponse = await fetch(server.tokenUrl, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+          },
+          body: urlEncodeForm({
+            grant_type: 'authorization_code',
+            client_id: server.clientId,
+            redirect_uri: server.callbackUrl,
+            code,
+            code_verifier: codeVerifier,
+          }),
+        });
+        const data = await tokenResponse.json();
+
+        resolve({...server, token: data.access_token});
+      } catch (e) {
+        reject(e);
+      }
+    });
+  };
+
+  function authenticate(server, usePkce = false) {
+    if (usePkce) {
+      return authPkceFlow(server);
+    } else {
+      return authImplicitFlow(server);
+    }
+  }
+
+  function getServerData(server, version) {
+    const serverDataVersion = {
+      1: server => ({
+        serverId: getKey(server.base_url || server.baseUrl),
+        clientId: server.consumer_key || server.consumerKey,
+        baseUrl: server.base_url || server.baseUrl,
```

Comment:
The camelcased versions are no longer relevant in indico 2.x: https://github.com/indico/indico/commit/e5c841725de